### PR TITLE
fix compilation for win 10 and vs2019

### PR DESCRIPTION
**Pull request by xueh2**
fix compilation for win 10

1) adjust boost cmake
2) remove set(Boost_NO_BOOST_CMAKE ON)
3) for windows, remove add_definitions( -DBOOST_NO_EXCEPTIONS )
4) a few adjust for win linkage

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -184,21 +184,27 @@ set(Boost_USE_STATIC_LIBS OFF)
 set(Boost_USE_MULTITHREADED ON)
 set(Boost_USE_STATIC_RUNTIME OFF)
 # necessary for Windows and RHEL <=6 systems
-set(Boost_NO_BOOST_CMAKE ON)
+# set(Boost_NO_BOOST_CMAKE ON)
```
Comment (dchansen) [inline CMakeLists.txt:187]: This should be removed completely

File: CMakeLists.txt [hunk 2]
```diff
@@ -184,21 +184,27 @@ set(Boost_USE_STATIC_LIBS OFF)
...
 
 if(WIN32)
-add_definitions( -DHAVE_SNPRINTF) #Python decides to overwrite snprintf if this is not defined, because ofcourse it does.
+  add_definitions( -DBOOST_ALL_NO_LIB )
```
Comment (dchansen) [inline CMakeLists.txt:190]: Why do we need this? Current Gadgetron compiles and runs under Windows 10 VS 2019

File: CMakeLists.txt [hunk 3]
```diff
@@ -184,21 +184,27 @@ set(Boost_USE_STATIC_LIBS OFF)
...
 # We actually only use system and thread explicitly, but they require linking in date_time and chrono
 if (WIN32)
-find_package(Boost 1.65.0 COMPONENTS system date_time chrono program_options filesystem timer exception REQUIRED )
+  set(BOOST_LIB_LIST system thread date_time chrono program_options filesystem timer iostreams)
 else()
-find_package(Boost 1.65.0 COMPONENTS system program_options filesystem timer REQUIRED )
+  set(BOOST_LIB_LIST system thread program_options filesystem timer iostreams)
 endif()
+
+find_package(Boost COMPONENTS ${BOOST_LIB_LIST} REQUIRED)
```
Comment (dchansen) [inline CMakeLists.txt:202]: We should keep the specific version requirement.

File: apps/gadgetron/CMakeLists.txt [hunk 4]
```diff
@@ -1,5 +1,6 @@
 configure_file(gadgetron_config.in gadgetron_config.h)
 
+include_directories(${PUGIXML_INCLUDE_DIRS})
```
Comment (dchansen) [inline apps/gadgetron/CMakeLists.txt:3]: This is not needed and will break CMakes automatic include dependency management.

File: apps/gadgetron/CMakeLists.txt [hunk 5]
```diff
@@ -69,7 +70,8 @@ target_link_libraries(gadgetron
         gadgetron_core
         gadgetron_toolbox_log
         boost
-        BLAS)
+        BLAS 
+        ${PUGIXML_LIBRARIES} )
```
Comment (dchansen) [inline apps/gadgetron/CMakeLists.txt:74]: Again, ${PUGIXML_LIBRARIES} is not required here, and even if it was, we should use a target based approach instead, similar to the rest of the current CMake

File: core/MessageID.h [hunk 6]
```diff
@@ -2,6 +2,10 @@
 
 #include <cstdint>
 
+#ifdef ERROR
+    #undef ERROR
```
Comment (dchansen) [inline core/MessageID.h:6]: Why is this needed?

File: gadgets/dicom/CMakeLists.txt [hunk 7]
```diff
@@ -83,6 +83,10 @@ target_link_libraries(gadgetron_dicom
     ISMRMRD::ISMRMRD
     ${GT_DICOM_LIBRARIES} )
 
+if (WIN32)
+    target_link_libraries(gadgetron_dicom netapi32.lib wsock32.lib WS2_32.Lib)
```
Comment (dchansen) [inline gadgets/dicom/CMakeLists.txt:87]: This should not be required at all, and will almost certainly break in the future.
Have you tried just using dcmtk from VCPKG? It should link to the dependencies automatically.

File: apps/gadgetron/CMakeLists.txt (hunk 4 above)
Comment (xueh2) [reply apps/gadgetron/CMakeLists.txt:3]: In windows, without this line, it will have including error.

File: apps/gadgetron/CMakeLists.txt (hunk 5 above)
Comment (xueh2) [reply apps/gadgetron/CMakeLists.txt:74]: same here. without this, it has linkage error. But how could I do the target based method?

File: core/MessageID.h (hunk 6 above)
Comment (xueh2) [reply core/MessageID.h:6]: VS 2019 has some header already having ERROR defined. Generating compiling error without undef it.

**Comment by dchansen** [comment]
The current master branch compiles with DCMTK on Windows 10 with VS 2019.
Install all dependencies except MKL through vcpkg and it just compiles.  Let me know if you have any problems and I can help you.
I'm closing this pull request.
